expr/functions.go

```diff
@@ -0,0 +1,54 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/brimsec/zq/zng"
+)
+
+type Function func([]NativeValue) (NativeValue, error)
+
+var ErrWrongArgc = errors.New("wrong number of arguments")
+var ErrBadArgument = errors.New("bad argument")
+
+var allFns = map[string]Function{
+	"Math.sqrt": mathSqrt,
+}
+
+func lookupFunction(name string) *Function {
+	fn, ok := allFns[name]
+	if ok {
+		return &fn
+	}
+	return nil
+}
```

Comment:
The Function zero value is nil, so you can simplify this to
```suggestion
func lookupFunction(name string) Function {
	return allFns[name]
}
```
or drop it entirely and just say `fn := allFns[node.Function]` in compileFunctionCall.